components/NavDrawer/NavDrawer.tsx

```diff
@@ -128,8 +128,8 @@ const DrawerTitle = React.forwardRef<
       {...props}
     />
 
-    <DrawerClose>
-      <X />
+    <DrawerClose data-testid="drawer-close">
+      <X data-testid="drawer-x-icon" />
```

Comment:
nit: you might be more descriptive with the testid for `<X />` . I'm thinking "drawer-close-icon"